rllib/agents/trainer.py

```diff
@@ -908,16 +909,24 @@ def evaluate(self) -> dict:
                 num_rounds = int(
                     math.ceil(self.config["evaluation_num_episodes"] /
                               self.config["evaluation_num_workers"]))
-                num_workers = len(self.evaluation_workers.remote_workers())
-                num_episodes = num_rounds * num_workers
-                for i in range(num_rounds):
-                    logger.info("Running round {} of parallel evaluation "
-                                "({}/{} episodes)".format(
-                                    i, (i + 1) * num_workers, num_episodes))
-                    ray.get([
-                        w.sample.remote()
-                        for w in self.evaluation_workers.remote_workers()
+                num_episodes = self.config["evaluation_num_episodes"]
+                num_episodes_done = 0
+                for round_ in range(num_rounds):
+                    left_to_do = num_episodes - num_episodes_done
+                    batches = ray.get([
+                        w.sample.remote() for i, w in enumerate(
+                            self.evaluation_workers.remote_workers())
+                        if i < left_to_do
                     ])
+                    if SampleBatch.EPS_ID in batches[0]:
```

Comment:
Can you comment this EPS_ID logic a little bit? I looked at the definition of SampleBatch.EPS_ID, and I am still clueless why we need special handling here.
Also, instead of checking [0], is it better (slightly more future proof) to flip the if and for here? like:

```
for b in batchs:
  if SampleBatch.EPS_ID in b:
    num_episodes_done += len(set(b[SampleBatch.EPS_ID]))
```

Reply:
done, clarified in the comment